var/spack/repos/builtin/packages/intel-oneapi-mpi/package.py

```diff
@@ -87,17 +85,19 @@ def libs(self):
         libs += find_libraries(['libmpicxx', 'libmpifort'], lib_dir)
         libs += find_libraries('libmpi', release_lib_dir)
         libs += find_system_libraries(['libdl', 'librt', 'libpthread'])
+
+        # Find libfabric for libmpi.so
+        if '+external-libfabric' in self.spec:
```

Comment:
There is no way to prevent the installer from installing libfabric. It goes into its own directory separate from the rest of the libs, so you can use an -L to control whether you use the libfabric that comes with mpi or a different one.

Reply:
If a package depends on `libfabric` and `intel-mpi`, I'm trying to make sure it doesn't get mixed up in terms of which `libfabric` implementation is used. I think your implementation of `libs` will be consistent, but underlying that there is still the issue of having two implementations of `libfabric` (if my understanding is correct, if a user installs `intel-oneapi-mpi+external-libfabric`, the MPI install still uses the "internal" libfabric installation - did I understand that directly?).

Reply:
Perhaps the easiest thing to start with is for me to confirm:

> if my understanding is correct, if a user installs intel-oneapi-mpi+external-libfabric, the MPI install still uses the "internal" libfabric installation - did I understand that directly?

Reply:
No. The libfabric used by mpi is found using LD_LIBRARY_PATH/RPATH. It is not a link-time dependence. My comment earlier about -L was incorrect.

Here is the layout on disk:
```
rscohn1@vs-sycl-3:ndarray$ ls /opt/intel/oneapi/mpi/latest/libfabric/lib/
libfabric.so  libfabric.so.1  prov
rscohn1@vs-sycl-3:ndarray$ ls /opt/intel/oneapi/mpi/latest/lib
debug         libmpicxx.so.12      libmpifort.so         libmpijava.so             libtvmpi.so         pkgconfig
debug_mt      libmpicxx.so.12.0    libmpifort.so.12      libmpijava.so.1           libtvmpi.so.12      release
libmpicxx.a   libmpicxx.so.12.0.0  libmpifort.so.12.0    libmpijava.so.1.0         libtvmpi.so.12.0.0  release_mt
libmpicxx.so  libmpifort.a         libmpifort.so.12.0.0  libmpi_shm_heap_proxy.so  mpi.jar
rscohn1@vs-sycl-3:ndarray$ ls /opt/intel/oneapi/mpi/latest/lib/release
libmpi.a    libmpi_ilp64.a   libmpi_ilp64.so.4    libmpi.so     libmpi.so.12.0
libmpi.dbg  libmpi_ilp64.so  libmpi_ilp64.so.4.1  libmpi.so.12  libmpi.so.12.0.0
rscohn1@vs-sycl-3:ndarray$
```

Reply:
OK I think I understand. FYI I'm assuming that `intel-mpi` will likely have constraints on what Spack-built `libfabric`  is used (i.e. if we use a new version of libfabric that is not compatible with an old installation of `intel-mpi`), but I think that can be handled in a later PR.

If the following looks good for a commit message, let me know and I will merge this:

```
Intel mpi: allow use of external libfabric

Intel mpi comes with an installation of libfabric (which it needs as a
dependency). It can use other implementations of libfabric at runtime
though, so if you install a package that depends on `mpi` and
`libfabric`, you can specify `intel-mpi+external-libfabric` and ensure
that the Spack-built instance is used (both by `intel-mpi` and the
root).

Apply analogous change to intel-oneapi-mpi.
```